apps/functions-subscription/src/adapters/azure/functions/process-trial-changes.ts

```diff
@@ -22,6 +32,24 @@ export const makeTrialChangesHandler =
         TE.traverseArray((trial) =>
           pipe(
             env.channelAdmin.create(trial.id),
+            TE.chainFirst(() =>
+              pipe(
+                env.activationJobReader.get(trial.id),
+                TE.flatMap(
+                  O.fold(
+                    () =>
+                      pipe(
+                        makeActivationJob({
+                          trialId: trial.id,
+                          usersToActivate: 0 as NonNegativeInteger,
+                        }),
+                        env.activationJobWriter.insert,
+                      ),
+                    TE.right,
+                  ),
+                ),
+              ),
+            ),
```

Comment:
**suggestion (non-blocking)**: You can skip the `env.activationJobReader.get` call and just use `env.activationJobWriter.insert`, taking care of the ItemAlreadyExists error if it comes up. This should simplify the code a little bit.

Reply:
it makes sense!